src/System.IdentityModel.Tokens/SymmetricSignatureProvider.cs

```diff
@@ -82,7 +82,19 @@ public SymmetricSignatureProvider(SymmetricSecurityKey key, string algorithm)
 
         public override bool IsSupportedAlgorithm(string algorithm)
         {
-            return false;
+            if (string.IsNullOrWhiteSpace(algorithm))
+                return false;
+
+            switch (algorithm)
+            {
+                case SecurityAlgorithms.HmacSha1Signature:
```

Comment:
We are dropping support for SHA1.  It is not supported in JWA https://tools.ietf.org/html/rfc7518#section-3 and is known to be at the end of its lifetime. If we do support it, it must be for signature validation only.